.github/workflows/test-cygwin.yml

```
@@ -0,0 +1,107 @@
+name: Test Cygwin
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  build:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-minor-version: [7, 8, 9]
+
+    timeout-minutes: 40
+
+    name: Python 3.${{ matrix.python-minor-version }}
+
+    steps:
+      - name: Fix line endings
+        run: |
+          git config --global core.autocrlf input
+
+      - name: Checkout Pillow
+        uses: actions/checkout@v3
+
+      - name: Install Cygwin
+        uses: cygwin/cygwin-install-action@v2
+        with:
+          platform: x86_64
+          packages: >
+            ImageMagick gcc-g++ ghostscript jpeg libfreetype-devel
+            libimagequant-devel libjpeg-devel liblapack-devel
+            liblcms2-devel libopenjp2-devel libraqm-devel
+            libtiff-devel libwebp-devel libxcb-devel libxcb-xinerama0
+            make netpbm perl
+            python3${{ matrix.python-minor-version }}-cffi
+            python3${{ matrix.python-minor-version }}-cython
+            python3${{ matrix.python-minor-version }}-devel
+            python3${{ matrix.python-minor-version }}-numpy
+            python3${{ matrix.python-minor-version }}-sip
+            python3${{ matrix.python-minor-version }}-tkinter
+            qt5-devel-tools subversion xorg-server-extra zlib-devel
+
+      - name: Add Lapack to PATH
+        uses: egor-tensin/cleanup-path@v1
+        with:
+          dirs: 'C:\cygwin\bin;C:\cygwin\lib\lapack'
+
+      - name: pip cache
+        uses: actions/cache@v3
+        with:
+          path: 'C:\cygwin\home\runneradmin\.cache\pip'
+          key: ${{ runner.os }}-cygwin-pip3.${{ matrix.python-minor-version }}-${{ hashFiles('.ci/install.sh') }}
+          restore-keys: |
+            ${{ runner.os }}-cygwin-pip3.${{ matrix.python-minor-version }}-
+
+      - name: Build system information
+        run: |
+          dash.exe -c "python3 .github/workflows/system-info.py"
+
+      - name: Install dependencies
+        run: |
+          bash.exe .ci/install.sh
+
+      - name: Install a different NumPy
+        shell: dash.exe -l "{0}"
+        run: |
+          python3 -m pip install -U 'numpy!=1.21.*'
+
+      - name: Build
+        shell: bash.exe -eo pipefail -o igncr "{0}"
+        run: |
+          bash.exe .ci/build.sh
```

```suggestion
        run: |
          bash.exe -eo pipefail -o igncr .ci/build.sh
```
Probably not necessary to invoke bash twice?
By omitting the shell param, it will be invoked via PowerShell, but at lesat it seems a little easier to read.

---

Alternately, we could do [like the Linux tests](https://github.com/python-pillow/Pillow/blob/9c580ce64902b23579e04a7890ba309438c152b2/.github/workflows/test.yml#L66-L68):
```suggestion
        shell: bash.exe -eo pipefail -o igncr "{0}"
        run: |
          .ci/build.sh
```
which may run into trouble if git doesn't set up the permissions properly.  Do you have a preference as to which goes in?

---

Your suggestion looks like the better option to me (assuming it works as expected).